### PR TITLE
Users/tedeshpa/node cosmos db

### PR DESCRIPTION
**Pull request by tejasd1990**

### CODE REVIEW

File: node/express/webappCosmosDB/Application/routes/databaseOperations.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+var MongoClient = require("mongodb").MongoClient;
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/express/webappCosmosDB/Application/routes/databaseOperations.js]: moved out.

File: node/express/webappCosmosDB/armTemplates/windows-webapp-template.json [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            }
+        },
+        {
+            "apiVersion": "2015-04-08",
+            "kind": "MongoDB",
+            "type": "Microsoft.DocumentDb/databaseAccounts",
+            "name": "[parameters('databaseAccountId')]",
+            "location": "[parameters('databaseAccountLocation')]",
+            "properties": {
+                "databaseAccountOfferType": "Standard",
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/express/webappCosmosDB/armTemplates/windows-webapp-template.json]: currently, cosmosdb supports only 'Standard' sku. Also, the spec doesn't contain it as an input parameter

File: node/plain/webappCosmosDB/Application/connectionData.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "connectionString": "connectionString",
+    "databaseName": "database23",
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/plain/webappCosmosDB/Application/connectionData.json]: changed to sampledatabase and samplecollection

File: node/plain/webappCosmosDB/Application/server.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    const collection = db.collection(collectionName);
+    // Insert some documents
+    collection.insertMany([
+        itemBody
+    ], function (err, result) {
+        assert.equal(err, null);
+        assert.equal(1, result.result.n);
+        assert.equal(1, result.ops.length);
+        console.log("Inserted 1 document into the collection");
+        callback();
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/plain/webappCosmosDB/Application/server.js]: done, now sending error and result to respective callbacks

File: node/plain/webappCosmosDB/Application/server.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+connectionString = ("mongodb://" + encodeURIComponent(userName) + ":" + encodeURIComponent(password) + "@" + stringSplit2[1]);
+
+function insertDocument(db, itemBody, callback) {
+    // Get the documents collection
+    const collection = db.collection(collectionName);
+    // Insert some documents
+    collection.insertMany([
+        itemBody
+    ], function (err, result) {
+        assert.equal(err, null);
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/plain/webappCosmosDB/Application/server.js]: removed assertions, replaced with appropriate responses

File: node/plain/webappCosmosDB/Application/server.js [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            client.close();
+        });
+    });
+}
+
+http.createServer(function (req, res) {
+
+    fs.readFile('index.html', function (err, data) {
+        res.writeHead(200, { 'Content-Type': 'text/html', 'Content-Length': data.length });
+        addRecord("index");
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/plain/webappCosmosDB/Application/server.js]: the client methods are not async, but they accept callbacks. corrected.

File: node/plain/webappCosmosDB/Application/server.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        assert.equal(err, null);
+        console.log(`Found ${docs.length} records`);
+        callback(docs.length);
+    });
+}
+
+/**
+ * Query the container using SQL
+ */
+function queryContainer(callback) {
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/plain/webappCosmosDB/Application/server.js]: done

File: node/plain/webappCosmosDB/Application/server.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        console.log("Inserted 1 document into the collection");
+        callback();
+    });
+}
+
+function findDocuments(db, callback) {
+    // Get the documents collection
+    const collection = db.collection(collectionName);
+    // Find some documents
+    collection.find({}).toArray(function (err, docs) {
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/plain/webappCosmosDB/Application/server.js]: using count()

File: node/express/webappCosmosDB/Application/routes/index.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+﻿'use strict';
+var express = require('express');
+var router = express.Router();
+var dbOperations = require('./databaseOperations.js');
+
+/* GET home page. */
+router.get('/', function (req, res) {
+    dbOperations.addRecord("index");
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/express/webappCosmosDB/Application/routes/index.js]: same as above

File: node/plain/webappCosmosDB/Application/Web.Debug.config [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    <customErrors mode="Off"/>
+  </system.web>
+
+  <system.webServer>
+    <!-- mimeMap enables IIS to serve particular file types as specified by fileExtension. -->
+    <staticContent>
+      <mimeMap fileExtension=".svg" mimeType="image/svg+xml" />
+    </staticContent>
+    
+    <iisnode loggingEnabled="true"
```
Comment (sachinma) [inline node/plain/webappCosmosDB/Application/Web.Debug.config]: You should use tis file from the node express and node plain projects in this repository.

File: node/express/webappCosmosDB/Application/databaseOperations.js [hunk 11]
```diff
@@ -16,61 +15,78 @@ connectionString = ("mongodb://" + encodeURIComponent(userName) + ":" + encodeUR
...
             itemBody
         ], function (err, result) {
-            assert.equal(err, null);
-            assert.equal(1, result.result.n);
-            assert.equal(1, result.ops.length);
-            console.log("Inserted 1 document into the collection");
+            if (err != null) {
+                errorCallback(err, 500)
+            }
+            if (result.ops.length == 1) {
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/express/webappCosmosDB/Application/databaseOperations.js]: was logging for testing. removed

File: node/express/webappCosmosDB/Application/databaseOperations.js [hunk 12]
```diff
@@ -16,61 +15,78 @@ connectionString = ("mongodb://" + encodeURIComponent(userName) + ":" + encodeUR
...
+                console.log("Inserted 1 document into the collection");
+            }
             callback();
         });
     },
 
-    findDocuments: function (db, callback) {
+    findDocuments: function (db, callback, errorCallback) {
```
[the remark this answers is not in the document]
Comment (tejasd1990) [reply node/express/webappCosmosDB/Application/databaseOperations.js]: merged the 2 functions

**Pull request merged**